api/api.go

```diff
@@ -213,41 +221,46 @@ func (nc *networkCreate) parseOptions() []libnetwork.NetworkOption {
 	return setFctList
 }
 
-func (ej *endpointJoin) parseOptions() []libnetwork.EndpointOption {
-	var setFctList []libnetwork.EndpointOption
+func (ej *sandboxCreate) parseOptions() []libnetwork.SandboxOption {
+	var setFctList []libnetwork.SandboxOption
 	if ej.HostName != "" {
-		setFctList = append(setFctList, libnetwork.JoinOptionHostname(ej.HostName))
+		setFctList = append(setFctList, libnetwork.OptionHostname(ej.HostName))
 	}
 	if ej.DomainName != "" {
-		setFctList = append(setFctList, libnetwork.JoinOptionDomainname(ej.DomainName))
+		setFctList = append(setFctList, libnetwork.OptionDomainname(ej.DomainName))
 	}
 	if ej.HostsPath != "" {
-		setFctList = append(setFctList, libnetwork.JoinOptionHostsPath(ej.HostsPath))
+		setFctList = append(setFctList, libnetwork.OptionHostsPath(ej.HostsPath))
 	}
 	if ej.ResolvConfPath != "" {
-		setFctList = append(setFctList, libnetwork.JoinOptionResolvConfPath(ej.ResolvConfPath))
+		setFctList = append(setFctList, libnetwork.OptionResolvConfPath(ej.ResolvConfPath))
 	}
 	if ej.UseDefaultSandbox {
-		setFctList = append(setFctList, libnetwork.JoinOptionUseDefaultSandbox())
+		setFctList = append(setFctList, libnetwork.OptionUseDefaultSandbox())
 	}
 	if ej.DNS != nil {
 		for _, d := range ej.DNS {
-			setFctList = append(setFctList, libnetwork.JoinOptionDNS(d))
+			setFctList = append(setFctList, libnetwork.OptionDNS(d))
 		}
 	}
 	if ej.ExtraHosts != nil {
 		for _, e := range ej.ExtraHosts {
-			setFctList = append(setFctList, libnetwork.JoinOptionExtraHost(e.Name, e.Address))
+			setFctList = append(setFctList, libnetwork.OptionExtraHost(e.Name, e.Address))
 		}
 	}
 	if ej.ParentUpdates != nil {
 		for _, p := range ej.ParentUpdates {
-			setFctList = append(setFctList, libnetwork.JoinOptionParentUpdate(p.EndpointID, p.Name, p.Address))
+			setFctList = append(setFctList, libnetwork.OptionParentUpdate(p.EndpointID, p.Name, p.Address))
```

Comment:
IMO, we should just drop this from REST api since this is all about supporting links for backward compatibility. No reason to expose that as an api. I know this PR didn't introduce it but...